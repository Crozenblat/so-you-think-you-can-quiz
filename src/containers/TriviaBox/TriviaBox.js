import React, { useState, useEffect, Fragment } from "react";
import styled from "styled-components";
import { CSSTransition } from "react-transition-group";

import axios from "../../axios-questions";

import withErrorHandler from "../../hoc/withErrorHandler/withErrorHandler";

import BoxWrapper from "../../components/UI/BoxWrapper/BoxWrapper";

import DifficultyButton from "../../components/UI/Buttons/DifficultyButton";
import SettingsEnterButton from "../../components/UI/Buttons/SettingsEnterButton";
import ChoiceButton from "../../components/UI/Buttons/ChoiceButton";
import SubmitButton from "../../components/UI/Buttons/SubmitButton";

import SettingsPanel from "../../components/SettingsPanel/SettingsPanel";

import AnimationWrapper from "../../components/UI/AnimationWrapper/AnimationWrapper";
import FinalScreen from "../../components/FinalScreen/FinalScreen";

import { mixUp } from "../../utilities/questionUtilities";


const Heading = styled.h1`
    margin-top: 3rem;
    font-size: 6.5rem;
    text-transform: capitalize;
    color: ${props => 
        props.correct === true && "green" ||
        props.correct === false && "red" ||
        props.correct === null && "white"
    };
`;

const Message = styled.p`
    text-transform: capitalize;
    font-size: ${props => props.hasError ?  "2rem" : "3rem"};
    color: ${props => props.hasError ? "red": "white"};
`;

const ChoiceForm = styled.form`
    display: flex;
    flex-direction: column;
`;

const TriviaBox = props => {
//Edit Header Content
    const [headingText, setHeadingText] = useState("So you think you can quiz!");
    const [messageText, setMessageText] = useState("Please select your difficulty below");
//Settings Panel Visibility
    const [settingsActive, setSettingsActive] = useState(false);
//Parameters to Fetch Questions
    const [questionAmount, setQuestionAmount] = useState(10);
    const [category, setCategory] = useState("category=9");
//Parameters to Serve Questions
    const [questionList, setQuestionList] = useState(null);
    const [questionNumber, setQuestionNumber] = useState(null);
    const [choices, setChoices] = useState(null);
    const [correctAnswer, setCorrectAnswer] = useState(null);
    const [selectedAnswer, setSelectedAnswer] = useState(null);
    const [isCorrect, setIsCorrect] = useState(null);
    const [score, setScore] = useState(0);
    const [isDisabled, setIsDisabled] = useState(null);
//Control Screen View
    const [view, setView] = useState("welcomeScreen");
    const [displayView, setDisplayView] = useState(false);
//Necessary in order to update the questionNumber before calling displayNextQuestion(); Acts as a callback to setQuestionNumber.
    useEffect(() => {
        setDisplayView(true);
        if(questionList !== null && questionNumber < questionList.length){
            displayNextQuestion(questionList[questionNumber]);
            setView("questionScreen");
        } else if(questionNumber !== null && questionNumber === questionList.length){
            props.colorChange();
            setView("finalScreen");
        }
    },[questionNumber]);

    const initializeQuestions = async (amount, category, difficulty) => {
        let questions = await axios.get(`?amount=${amount}&${category}&difficulty=${difficulty}&type=multiple&encode=url3986`);
        setQuestionList(questions.data.results);
        setQuestionNumber(0);
    };

    const displayNextQuestion = (newQuestion) => {
        props.colorChange();
        setChoices((mixUp([newQuestion.correct_answer, ...newQuestion.incorrect_answers])));
        setCorrectAnswer(decodeURIComponent(newQuestion.correct_answer));
        setIsCorrect(null);
        setHeadingText(`Question ${questionNumber + 1}`);
        setMessageText(decodeURIComponent(newQuestion.question));
    };

    const checkAnswer = () => {
        if(selectedAnswer === correctAnswer){
            props.colorChange("linear-gradient(to left,#56AB2F,#A8E063)");
            props.undulatePanels();
            questionList[questionNumber].correct = true;
            setIsCorrect(true);
            setHeadingText("Correct!");
            setScore(prev => prev + 1);
        } else if(selectedAnswer !== correctAnswer){
            props.colorChange("linear-gradient(to left,#ED213A,#93291E)");
            questionList[questionNumber].correct = false;
            questionList[questionNumber].selected = selectedAnswer;
            setIsCorrect(false);
            setHeadingText("Wrong");
        }
        setSelectedAnswer(null);
        setIsDisabled(true);
        setTimeout(() => {
            setDisplayView(false);
//Throws a memory leak warning during testing for setting state on an unmounted component. Necessary for animation to work.
            setTimeout(() => {
                setIsDisabled(false);
                setQuestionNumber(prev => prev + 1);
            }, 400);
        }, 2000);
    }
    
    let screen;

    const boxText = <Fragment>
                        <Heading correct={isCorrect}>{headingText}</Heading>
                        <Message>{messageText}</Message>
                    </Fragment>

    switch(view){
        case "welcomeScreen":
            screen = <AnimationWrapper>
                        {boxText}
                        <div>
                            {["Easy", "Medium", "Hard"].map((buttonLabel, index) => {
                                return <DifficultyButton 
                                        key={index}
                                        difficulty={buttonLabel.toLowerCase()} 
                                        clicked={(difficulty) => {
                                            initializeQuestions(questionAmount, category, difficulty);
                                            setDisplayView(false);
                                        }}
                                        >{buttonLabel}</DifficultyButton>
                            })}
                        </div>
                        <SettingsEnterButton setActiveTrue={() => setSettingsActive(true)}>Settings</SettingsEnterButton>
                        <SettingsPanel 
                            active={settingsActive} 
                            setActiveFalse={() => setSettingsActive(false)} 
                            questionAmount={questionAmount}
                            setQuestionAmount={setQuestionAmount}
                            setCategory={setCategory}
                            />
                        <div>
                            <p>Favicon made by <a href="https://www.flaticon.com/authors/roundicons" title="Roundicons">Roundicons</a> from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a></p>
                            <p>Icons made by <a href="https://www.flaticon.com/authors/smashicons" title="Smashicons">Smashicons</a> & <a href="https://www.flaticon.com/authors/pixel-perfect" title="Pixel perfect">Pixel Perfect</a> from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a></p>
                        </div>
                     </AnimationWrapper>
            break;
        case "questionScreen":
            let choiceButtons = choices.map((option) => <ChoiceButton key={option} disabled={isDisabled} isCorrect={isCorrect} correct={correctAnswer} value={option} clicked={(choice) => setSelectedAnswer(decodeURIComponent(choice))}>{option}</ChoiceButton>);
            screen = <AnimationWrapper>
                        {boxText}
                        <ChoiceForm>
                            {choiceButtons}
                            <SubmitButton isDisabled={!selectedAnswer} clicked={(e) => {e.preventDefault(); checkAnswer();}}>Submit</SubmitButton>
                        </ChoiceForm>
                     </AnimationWrapper>
            break;
        case "finalScreen":
            screen = <AnimationWrapper>
                        <FinalScreen
                            setKey={props.setKey}
                            setDisplayView={setDisplayView}
                            score={score}
                            colorChange={props.colorChange}
                            questionAmount={questionList.length}
                            questionList={questionList}
                        />
                     </AnimationWrapper>
            break;
    }

    return(
        <BoxWrapper>
            <CSSTransition 
            unmountOnExit
            in={displayView}
            timeout={400}
            classNames='slide'
            appear> 
                {screen}
            </CSSTransition>
        </BoxWrapper>

    );
};

export default withErrorHandler(TriviaBox, axios);