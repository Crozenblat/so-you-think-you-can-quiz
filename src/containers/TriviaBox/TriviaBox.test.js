import React from "react";
import mockAxios from "../../axios-questions";
import { render, screen, fireEvent} from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';

import TriviaBox from "./TriviaBox";

jest.mock("../../axios-questions");

describe("Starting Screen", () => {

    it("should render correctly", () => {
        const wrapper = render(<TriviaBox/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("should reveal the settings panel when the settings entry button is clicked", () => {
        render(<TriviaBox/>);
        const settingsBtn = screen.getByRole("button", {name: /settings/i});
        const settingsPnl = screen.getByTestId("settings-panel");

        expect(settingsPnl).toHaveStyle("height: 0%");

        fireEvent.click(settingsBtn);

        expect(settingsPnl).toHaveStyle("height: 100%")
    });
});

describe("Question Screen", () => {

    const mockColorChange = jest.fn();
    const mockUndulatePanels = jest.fn();

    afterEach(() => {
        jest.clearAllMocks();
    });

    it("should render the first question when difficulty button is clicked", async () => {
        render(<TriviaBox colorChange={mockColorChange} undulatePanels={mockUndulatePanels}/>);
        const diffBtn = screen.getByRole("button", {name: /easy/i});
        fireEvent.click(diffBtn);

        const heading = await screen.findByText("Question 1");
        const question = await screen.findByText("Which best selling toy of 1983 caused hysteria, resulting in riots breaking out in stores?");
        const choices = await screen.findAllByRole("radio");
        const submit = await screen.findByRole("button", {name: /submit/i});

        expect(heading).toBeInTheDocument();
        expect(question).toBeInTheDocument();
        expect(submit).toBeInTheDocument();
        expect(choices).toHaveLength(4);
        expect(choices.map(choice => choice.id)).toEqual(expect.arrayContaining(["Cabbage Patch Kids", "Transformers", "Care Bears", "Rubikâ€™s Cube"]));

        expect(mockColorChange).toHaveBeenCalled();
    });

    it("should play the 'correct' animation when question is answered correctly", async () => {
        render(<TriviaBox colorChange={mockColorChange} undulatePanels={mockUndulatePanels}/>);
        const diffBtn = screen.getByRole("button", {name: /easy/i});
        fireEvent.click(diffBtn);

        const correctChoice = await screen.findByText("Cabbage Patch Kids");
        const submit = await screen.findByRole("button", {name: /submit/i});

        fireEvent.click(correctChoice);
        fireEvent.click(submit);

        const heading = await screen.findByText("Correct!");

        expect(heading).toHaveStyle("color: green");
        expect(mockColorChange).toHaveBeenCalledWith("linear-gradient(to left,#56AB2F,#A8E063)");
        expect(mockUndulatePanels).toHaveBeenCalled();
    });

    it("should play the 'incorrect' animation when question is answered incorrectly", async () => {
        render(<TriviaBox colorChange={mockColorChange} undulatePanels={mockUndulatePanels}/>);
        const diffBtn = screen.getByRole("button", {name: /easy/i});
        fireEvent.click(diffBtn);

        const correctChoice = await screen.findByText("Cabbage Patch Kids");
        const incorrectChoice = await screen.findByText("Transformers");
        const submit = await screen.findByRole("button", {name: /submit/i});

        fireEvent.click(incorrectChoice);
        fireEvent.click(submit);

        const heading = await screen.findByText("Wrong");

        expect(heading).toHaveStyle("color: red");
        expect(correctChoice).toHaveStyle("background: green");
        expect(mockColorChange).toHaveBeenCalledWith("linear-gradient(to left,#ED213A,#93291E)");
    });

    it("should display the next question when the current question is answered", async () => {
        render(<TriviaBox colorChange={mockColorChange} undulatePanels={mockUndulatePanels}/>);
        const diffBtn = screen.getByRole("button", {name: /easy/i});
        fireEvent.click(diffBtn);

        let firstChoice = await screen.findByText("Transformers");
        let submit = await screen.findByRole("button", {name: /submit/i});

        fireEvent.click(firstChoice);
        fireEvent.click(submit);     

        const heading = await screen.findByText("Question 2");
        const question = await screen.findByText("Area 51 is located in which US state?");
        const choices = await screen.findAllByRole("radio");
        submit = await screen.findByRole("button", {name: /submit/i});


        expect(heading).toBeInTheDocument();
        expect(heading).toHaveStyle("color: white");
        expect(question).toBeInTheDocument();
        expect(submit).toBeInTheDocument();
        expect(choices).toHaveLength(4);
        expect(choices.map(choice => choice.id)).toEqual(expect.arrayContaining(["Nevada", "Arizona", "New Mexico", "Utah"]));

        expect(mockColorChange).toHaveBeenCalledTimes(3);
    });

    it("should display the final screen after the last question is answered", async () => {
        jest.setTimeout(7000);

        render(<TriviaBox colorChange={mockColorChange} undulatePanels={mockUndulatePanels}/>);
        const diffBtn = screen.getByRole("button", {name: /easy/i});
        fireEvent.click(diffBtn);

        let correctChoice = await screen.findByText("Cabbage Patch Kids");
        let submit = await screen.findByRole("button", {name: /submit/i});

        fireEvent.click(correctChoice);
        fireEvent.click(submit);     

        correctChoice = await screen.findByText("Nevada");
        submit = await screen.findByRole("button", {name: /submit/i});

        fireEvent.click(correctChoice);
        fireEvent.click(submit);
        
        const playAgain = await screen.findByRole("button", {name: /Play Again/i});
        expect(playAgain).toBeInTheDocument();
    });
});