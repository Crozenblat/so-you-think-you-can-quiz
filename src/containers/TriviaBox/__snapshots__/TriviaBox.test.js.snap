// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Starting Screen should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm cMjvdi"
      >
        <div
          class="sc-fzokOt iqMaKF slide-enter slide-enter-active"
        >
          <h1
            class="sc-fznWqX geNzE"
          >
            So you think you can quiz!
          </h1>
          <p
            class="sc-fzoiQi lkEGKF"
          >
            Please select your difficulty below
          </p>
          <div>
            <button
              class="sc-AxiKw hhNwSg"
            >
              Easy
            </button>
            <button
              class="sc-AxiKw fnWvaT"
            >
              Medium
            </button>
            <button
              class="sc-AxiKw bvltNc"
            >
              Hard
            </button>
          </div>
          <button
            class="sc-AxhCb comPFF"
          >
            Settings
          </button>
          <div
            class="sc-fznZeY cXSVUF"
            data-testid="settings-panel"
          >
            <h1
              class="sc-fznKkj jPpNVj"
            >
              Categories
            </h1>
            <form
              action="#"
            >
              <input
                class="sc-Axmtr gifGgb"
                id="0"
                name="category"
                type="radio"
                value="category=31"
              />
              <label
                class="sc-AxmLO jxwHeO"
                for="0"
              >
                Anime & Manga
              </label>
              <input
                class="sc-Axmtr gifGgb"
                id="1"
                name="category"
                type="radio"
                value="category=10"
              />
              <label
                class="sc-AxmLO jxwHeO"
                for="1"
              >
                Books
              </label>
              <input
                class="sc-Axmtr gifGgb"
                id="2"
                name="category"
                type="radio"
                value="category=18"
              />
              <label
                class="sc-AxmLO jxwHeO"
                for="2"
              >
                Computers
              </label>
              <input
                class="sc-Axmtr gifGgb"
                id="3"
                name="category"
                type="radio"
                value="category=11"
              />
              <label
                class="sc-AxmLO jxwHeO"
                for="3"
              >
                Film
              </label>
              <input
                class="sc-Axmtr gifGgb"
                id="4"
                name="category"
                type="radio"
                value="category=22"
              />
              <label
                class="sc-AxmLO jxwHeO"
                for="4"
              >
                Geography
              </label>
              <input
                class="sc-Axmtr gifGgb"
                id="5"
                name="category"
                type="radio"
                value="category=23"
              />
              <label
                class="sc-AxmLO jxwHeO"
                for="5"
              >
                History
              </label>
              <input
                class="sc-Axmtr gifGgb"
                id="6"
                name="category"
                type="radio"
                value="category=12"
              />
              <label
                class="sc-AxmLO jxwHeO"
                for="6"
              >
                Music
              </label>
              <input
                class="sc-Axmtr gifGgb"
                id="7"
                name="category"
                type="radio"
                value="category=17"
              />
              <label
                class="sc-AxmLO jxwHeO"
                for="7"
              >
                Science & Nature
              </label>
              <input
                class="sc-Axmtr gifGgb"
                id="8"
                name="category"
                type="radio"
                value="category=14"
              />
              <label
                class="sc-AxmLO jxwHeO"
                for="8"
              >
                Television
              </label>
              <input
                class="sc-Axmtr gifGgb"
                id="9"
                name="category"
                type="radio"
                value="category=15"
              />
              <label
                class="sc-AxmLO jxwHeO"
                for="9"
              >
                Video Games
              </label>
              <input
                checked=""
                class="sc-Axmtr gifGgb"
                id="10"
                name="category"
                type="radio"
                value="category=9"
              />
              <label
                class="sc-AxmLO jxwHeO"
                for="10"
              >
                General Knowledge
              </label>
            </form>
            <label
              class="sc-fzplWN ihqSyJ"
              for="question-amount"
            >
              Number of Questions:
            </label>
            <input
              class="sc-fzpans hHctQZ"
              data-testid="question-amount"
              id="question-amount"
              type="number"
              value="10"
            />
            <img
              alt="Up Arrow"
              class="sc-fzozJi hEbiXo"
              src="[object Object]"
            />
            <img
              alt="Down Arrow"
              class="sc-fzoLsD gufRLf"
              src="[object Object]"
            />
            <button
              class="sc-fznyAO fRcocx"
              data-testid="settings-exit-button"
            >
              X
            </button>
          </div>
          <div>
            <p>
              Favicon made by 
              <a
                href="https://www.flaticon.com/authors/roundicons"
                title="Roundicons"
              >
                Roundicons
              </a>
               from 
              <a
                href="https://www.flaticon.com/"
                title="Flaticon"
              >
                www.flaticon.com
              </a>
            </p>
            <p>
              Icons made by 
              <a
                href="https://www.flaticon.com/authors/smashicons"
                title="Smashicons"
              >
                Smashicons
              </a>
               & 
              <a
                href="https://www.flaticon.com/authors/pixel-perfect"
                title="Pixel perfect"
              >
                Pixel Perfect
              </a>
               from 
              <a
                href="https://www.flaticon.com/"
                title="Flaticon"
              >
                www.flaticon.com
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm cMjvdi"
    >
      <div
        class="sc-fzokOt iqMaKF slide-enter slide-enter-active"
      >
        <h1
          class="sc-fznWqX geNzE"
        >
          So you think you can quiz!
        </h1>
        <p
          class="sc-fzoiQi lkEGKF"
        >
          Please select your difficulty below
        </p>
        <div>
          <button
            class="sc-AxiKw hhNwSg"
          >
            Easy
          </button>
          <button
            class="sc-AxiKw fnWvaT"
          >
            Medium
          </button>
          <button
            class="sc-AxiKw bvltNc"
          >
            Hard
          </button>
        </div>
        <button
          class="sc-AxhCb comPFF"
        >
          Settings
        </button>
        <div
          class="sc-fznZeY cXSVUF"
          data-testid="settings-panel"
        >
          <h1
            class="sc-fznKkj jPpNVj"
          >
            Categories
          </h1>
          <form
            action="#"
          >
            <input
              class="sc-Axmtr gifGgb"
              id="0"
              name="category"
              type="radio"
              value="category=31"
            />
            <label
              class="sc-AxmLO jxwHeO"
              for="0"
            >
              Anime & Manga
            </label>
            <input
              class="sc-Axmtr gifGgb"
              id="1"
              name="category"
              type="radio"
              value="category=10"
            />
            <label
              class="sc-AxmLO jxwHeO"
              for="1"
            >
              Books
            </label>
            <input
              class="sc-Axmtr gifGgb"
              id="2"
              name="category"
              type="radio"
              value="category=18"
            />
            <label
              class="sc-AxmLO jxwHeO"
              for="2"
            >
              Computers
            </label>
            <input
              class="sc-Axmtr gifGgb"
              id="3"
              name="category"
              type="radio"
              value="category=11"
            />
            <label
              class="sc-AxmLO jxwHeO"
              for="3"
            >
              Film
            </label>
            <input
              class="sc-Axmtr gifGgb"
              id="4"
              name="category"
              type="radio"
              value="category=22"
            />
            <label
              class="sc-AxmLO jxwHeO"
              for="4"
            >
              Geography
            </label>
            <input
              class="sc-Axmtr gifGgb"
              id="5"
              name="category"
              type="radio"
              value="category=23"
            />
            <label
              class="sc-AxmLO jxwHeO"
              for="5"
            >
              History
            </label>
            <input
              class="sc-Axmtr gifGgb"
              id="6"
              name="category"
              type="radio"
              value="category=12"
            />
            <label
              class="sc-AxmLO jxwHeO"
              for="6"
            >
              Music
            </label>
            <input
              class="sc-Axmtr gifGgb"
              id="7"
              name="category"
              type="radio"
              value="category=17"
            />
            <label
              class="sc-AxmLO jxwHeO"
              for="7"
            >
              Science & Nature
            </label>
            <input
              class="sc-Axmtr gifGgb"
              id="8"
              name="category"
              type="radio"
              value="category=14"
            />
            <label
              class="sc-AxmLO jxwHeO"
              for="8"
            >
              Television
            </label>
            <input
              class="sc-Axmtr gifGgb"
              id="9"
              name="category"
              type="radio"
              value="category=15"
            />
            <label
              class="sc-AxmLO jxwHeO"
              for="9"
            >
              Video Games
            </label>
            <input
              checked=""
              class="sc-Axmtr gifGgb"
              id="10"
              name="category"
              type="radio"
              value="category=9"
            />
            <label
              class="sc-AxmLO jxwHeO"
              for="10"
            >
              General Knowledge
            </label>
          </form>
          <label
            class="sc-fzplWN ihqSyJ"
            for="question-amount"
          >
            Number of Questions:
          </label>
          <input
            class="sc-fzpans hHctQZ"
            data-testid="question-amount"
            id="question-amount"
            type="number"
            value="10"
          />
          <img
            alt="Up Arrow"
            class="sc-fzozJi hEbiXo"
            src="[object Object]"
          />
          <img
            alt="Down Arrow"
            class="sc-fzoLsD gufRLf"
            src="[object Object]"
          />
          <button
            class="sc-fznyAO fRcocx"
            data-testid="settings-exit-button"
          >
            X
          </button>
        </div>
        <div>
          <p>
            Favicon made by 
            <a
              href="https://www.flaticon.com/authors/roundicons"
              title="Roundicons"
            >
              Roundicons
            </a>
             from 
            <a
              href="https://www.flaticon.com/"
              title="Flaticon"
            >
              www.flaticon.com
            </a>
          </p>
          <p>
            Icons made by 
            <a
              href="https://www.flaticon.com/authors/smashicons"
              title="Smashicons"
            >
              Smashicons
            </a>
             & 
            <a
              href="https://www.flaticon.com/authors/pixel-perfect"
              title="Pixel perfect"
            >
              Pixel Perfect
            </a>
             from 
            <a
              href="https://www.flaticon.com/"
              title="Flaticon"
            >
              www.flaticon.com
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
